universal-application-tool-0.0.1/app/controllers/LoginController.java

```diff
@@ -52,6 +59,25 @@ public Result idcsLoginWithRedirect(Http.Request request, Optional<String> redir
     return login(request, idcsClient).addingToSession(request, "redirectTo", redirectTo.get());
   }
 
+  public Result register(Http.Request request) {
+    String registerUrl = null;
+    try {
+      registerUrl = config.getString("idcs.register_uri");
+    } catch (ConfigException.Missing e) {
+      // leave it as null / empty.
+    }
+    if (Strings.isNullOrEmpty(registerUrl)) {
+      return badRequest("Registration is not enabled.");
+    }
+    // Redirect to the registration URL - then, when the user visits the site again, automatically
+    // log them in.
+    return redirect(registerUrl)
+        .addingToSession(
+            request,
+            "redirectTo",
+            routes.LoginController.idcsLoginWithRedirect(Optional.empty()).url());
```

Comment:
Nit: `WithRedirect` seems unnecessary.
```suggestion
            routes.LoginController.idcsLogin().url());
```

Reply:
Wouldn't be, but that one is not routed, actually.  `routes` can't handle distinguishing based on the presence or absence of a query param.